nep-dapi.mediawiki

```diff
@@ -0,0 +1,586 @@
+function getProvider(): Promise<Provider>
+</pre>
+
+'''Input arguments'''
+
+None
+
+'''Success return value'''
+
+object - Provider information defined by the interface below:
+<pre>
+interface Provider {
+  name: string;
+  website: string;
+  version: string;
+  compatibility: string[];
+}
+</pre>
+
+The compatibility field will return a list of strings that corresponds to NEPs that the dAPI conforms to. For example:
+<pre>
+compatibility: [
+    'NEP-14',
+    'NEP-23',
+    'NEP-29'
+]
+</pre>
+
+'''Error return value'''
+
+string - `NO_PROVIDER`|`CONNECTION_DENIED`
+
+'''Example'''
+
+<pre>
+getProvider()
+.then((provider: Provider) => {
+  const {
+    name,
+    website,
+    version,
+    compatibility,
+  } = provider;
+
+  console.log('Provider name: ' + name);
+  console.log('Provider website: ' + website);
+  console.log('Provider dAPI version: ' + version);
+  console.log('Provider dAPI compatibility: ' + JSON.stringify(compatibility));
+})
+.catch((err: string) => {
+  switch(err) {
+    case NO_PROVIDER:
+      console.log('No provider available.');
+      break;
+    case CONNECTION_DENIED:
+      console.log('The user rejected the request to connect with your dApp.');
+      break;
+  }
+});
+</pre>
+
+====getAccount====
+Return the Account that is currently connected to the dApp.
+
+'''Method Interface'''
+
+<pre>
+function getAccount(): Promise<{ address: string, publicKey: string }>
+</pre>
+
+'''Input arguments'''
+
+None
+
+'''Success return value'''
+
+address: string - Address of the connected account
+
+publicKey: string - Address of the connected account
+
+'''Error return value'''
+
+string - `NO_PROVIDER`|`CONNECTION_DENIED`
+
+'''Example'''
+
+<pre>
+getAccount()
+.then((account: Account) => {
+  const {
+    address,
+    publicKey,
+  } = account;
+
+  console.log('Provider address: ' + address);
+  console.log('Provider public key: ' + publicKey);
+})
+.catch((err: string) => {
+  switch(err) {
+    case NO_PROVIDER:
+      console.log('No provider available.');
+      break;
+    case CONNECTION_DENIED:
+      console.log('The user rejected the request to connect with your dApp');
+      break;
+  }
+});
+</pre>
+
+
+
+====getBalance====
+Return balance of a specific asset for the given account.
+
+'''Method Interface'''
+
+<pre>
+function getBalance({ address: string, asset: string }): Promise<{ address: string, asset: string, amount: string }}>
+</pre>
+
+'''Input arguments'''
+
+address: string - The asset symbol the check the balance for the connected account
+
+asset: string - The asset symbol the check the balance for the connected account
+
+
+'''Success return value'''
+
+address: string - Address of the connected account
+
+asset: string - Asset symbol for the requested balance
+
+amount: string - Parsed balance of the requested asset
+
+
+'''Error return value'''
+
+string - `NO_PROVIDER`|`CONNECTION_DENIED`
+
+'''Example'''
+
+<pre>
+getBalance({
+  address: 'AeysVbKWiLSuSDhg7DTzUdDyYYKfgjojru',
```

Comment:
I think that `getBalance` should support fetching of multiple balances (neon-js supports for this NEP5 tokens)

Reply:
Also, perhaps if address is not passed it should use the currently logged in one?

Reply:
Good idea! Also was thinking about making the asset optional. In the case that it is no present, the wallet will return all assets owned by this address. But curious to see what others opinion are on the matter.

Reply:
Updated to support multiple accounts and all assets